### PR TITLE
Feature/476 metering point connected

### PR DESCRIPTION
**Pull request by ZavezX**
### .NET

- Refactor existing method to parse ServiceBus metadata to `EventMetadata` object with added properties stated in [ADR-0008](https://github.com/Energinet-DataHub/green-energy-hub/blob/main/docs/architecture-decision-record/ADR-0008%20Integration%20events.md)
- Add common method to construct `EventHubMetadata` dictionary
- Add unit tests for added functionality surrounding parsing of ServiceBus meta data
- Change internal meta data for EventHub to include more values to be used in python part

### Python

- Add `MeteringPointConnected` class to be able to update existing periods for MeteringPoint
- Add unit tests for `period_mutation` functions
- Refactor columns for `Events` table corresponding to changes in .Net part for internal EventHub meta data

## References

* #476 
* #317

### CODE REVIEW

**Review by johevemi — APPROVED** [review]

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+namespace Energinet.DataHub.Aggregations.Application.Extensions
+{
+    public static class EnumExtension
+    {
+        /// <summary>
+        /// Gets the description attribute from an enum
+        /// </summary>
+        /// <param name="enumValue"></param>
+        /// <returns>string</returns>
+        public static string GetDescription(this Enum enumValue)
```
Comment (jonasdmoeller) [inline source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs:27]: If we use an enum for ConnectionState in MeteringPointConnectedEvent, this should not be necessary as we use custom converters for the json serialization.

File: source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py [hunk 2]
```diff
@@ -18,11 +18,16 @@
 def process_eventhub_item(df, epoch_id, events_delta_path):
     if len(df.head(1)) > 0:
         # Extract metadata from the eventhub message and wrap into containing dataframe
-        jsonDataFrame = df.select((df.properties["Id"]).alias("Id"), (df.properties["SchemaType"]).alias("type"), (df.body.cast(StringType()).alias("body")))
+        jsonDataFrame = df.select(
+            (df.properties["event_id"].alias("event_id")),
+            (df.properties["processed_date"].alias("processed_date")),
+            (df.properties["event_name"].alias("event_name")),
+            (df.properties["domain"].alias("domain")),
```
Comment (jonasdmoeller) [inline source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py:25]: Maybe we should move these json property names to a constant, as they are use several places.

File: source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py [hunk 3]
```diff
@@ -18,11 +18,16 @@
...
+            (df.body.cast(StringType()).alias("body")))
 
         # Append event
         jsonDataFrame.write \
-            .partitionBy("type") \
+            .partitionBy("event_name") \
```
Comment (jonasdmoeller) [inline source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py:30]: Same as previous comment

File: source/databricks/geh_stream/streaming_utils/events_data_lake_listener.py [hunk 4]
```diff
@@ -20,7 +20,7 @@
 def incomming_event_handler(df, epoch_id):
     if len(df.head(1)) > 0:
         for row in df.rdd.collect():
-            event_class = message_registry.get(row["type"])
+            event_class = message_registry.get(row["event_name"])
```
Comment (jonasdmoeller) [inline source/databricks/geh_stream/streaming_utils/events_data_lake_listener.py:23]: Same as previous comment

**Review by jonasdmoeller — CHANGES_REQUESTED** [review]

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs [hunk 5]
```diff
@@ -12,12 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System.ComponentModel;
+
 namespace Energinet.DataHub.Aggregations.Domain
 {
     public enum ConnectionState
     {
+        [Description("D03")]
```
[the remark this answers is not in the document]
Comment (mknic) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs:21]: It's necessary as we use the `Description` as the value on the python side of it. @ZavezX might be able to elaborate on this.

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs [hunk 6]
```diff
@@ -21,7 +21,7 @@ namespace Energinet.DataHub.Aggregations.Application.IntegrationEvents.MeteringP
     public record MeteringPointConnectedEvent(
 #pragma warning disable SA1313
             string MeteringPointId,
-            string GsrnNumber,
+            string ConnectionState,
```
[the remark this answers is not in the document]
Comment (mknic) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs:24]: I thought we agreed on using strings in our internal contract?

File: source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py (hunk 2 above)
Comment (mknic) [reply source/databricks/geh_stream/streaming_utils/eventhub_ingestor.py:25]: Good idea!

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ZavezX) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs:21]: The `Description` attribute is meant to be used to hold the business value, thus we as developers don't need to remember all the different codes, instead we can just extract it from the attribute.

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ZavezX) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs:24]: I also remember we agreed to use string for ConnectionString in our internal event model.

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs (hunk 1 above)
Comment (ZavezX) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs:27]: We use the `Description` attribute directly in the mapper for `MeteringPointConnected`, so we don't need to do custom json serialization as the `ConnectionState` property is a string.

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (jonasdmoeller) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/IntegrationEvents/MeteringPoints/MeteringPointConnectedEvent.cs:24]: We did agree on using strings in our internal contract. And using custom converters in the JsonSerializer we convert these enums to strings in a strongly typed manner, and thus preventing an invalid event being transfered through our ingestion layer.
If we still decide to use string properties in the event model, we should refactor the ConsumptionMeteringPointCreatedEvent to do the same.

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (jonasdmoeller) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Domain/ConnectionState.cs:21]: See comment below

File: source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs (hunk 1 above)
Comment (jonasdmoeller) [reply source/IntegrationEventListener/source/Energinet.DataHub.Aggregations.Application/Extensions/EnumExtension.cs:27]: See comment above

**Review by jonasdmoeller — APPROVED** [review]

**Pull request merged**